### ISSUE TITLE
💅 false positive noStaticOnlyClass 

### ISSUE THREAD

**Issue by kyptov** [biomejs/biome#3612]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.16.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  false
  All:                          false
  Enabled rules:
  complexity/noBannedTypes
  suspicious/noCatchAssign
  suspicious/noPrototypeBuiltins
  complexity/useLiteralKeys
  suspicious/noMisleadingInstantiator
  suspicious/noDebugger
  complexity/noMultipleSpacesInRegularExpressionLiterals
  complexity/noUselessLoneBlockStatements
  complexity/noVoid
  suspicious/noGlobalAssign
  suspicious/noUnsafeNegation
  suspicious/useValidTypeof
  correctness/noConstantCondition
  suspicious/noDuplicateParameters
  correctness/noEmptyPattern
  suspicious/noExtraNonNullAssertion
  complexity/noUselessTernary
  style/useConst
  suspicious/noCompareNegZero
  suspicious/noSelfCompare
  correctness/noUnreachableSuper
  security/noGlobalEval
  style/noNonNullAssertion
  correctness/noConstAssign
  correctness/noUnusedVariables
  correctness/useIsNan
  suspicious/noAsyncPromiseExecutor
  correctness/noSwitchDeclarations
  suspicious/noControlCharactersInRegex
  complexity/noUselessTypeConstraint
  style/noVar
  suspicious/noDoubleEquals
  style/useConsistentArrayType
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  suspicious/noRedeclare
  correctness/noInvalidConstructorSuper
  suspicious/noDuplicateObjectKeys
  suspicious/noFallthroughSwitchClause
  correctness/noUnreachable
  suspicious/noConfusingVoidType
  complexity/useOptionalChain
  suspicious/noDuplicateCase
  complexity/useRegexLiterals
  correctness/noSelfAssign
  style/useShorthandFunctionType
  suspicious/noEmptyBlockStatements
  suspicious/noShadowRestrictedNames
  complexity/noUselessCatch
  correctness/noInvalidUseBeforeDeclaration
  correctness/noUnsafeFinally
  style/useBlockStatements
  complexity/noUselessRename
  complexity/noUselessConstructor
  style/useEnumInitializers
  style/noDefaultExport
  complexity/noStaticOnlyClass
  style/useSingleVarDeclarator
  suspicious/noImportAssign
  style/noCommaOperator
  suspicious/useDefaultSwitchClauseLast
  style/noNamespace
  correctness/noGlobalObjectCalls
  correctness/noNewSymbol
  correctness/noEmptyCharacterClassInRegex
  suspicious/noClassAssign
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0
```

### Rule name

noStaticOnlyClass

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YQBiAHMAdAByAGEAYwB0ACAAYwBsAGEAcwBzACAARQBuAHQAaQB0AHkAPABUAD4AIAB7AAoAIAAgAHAAcgBvAHQAZQBjAHQAZQBkACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAcAByAGkAdgBhAHQAZQAgAHIAZQBhAGQAbwBuAGwAeQAgAHAAcgBvAHAAcwA6ACAAVAApACAAewB9AAoACgAgACAAZwBlAHQAIABwAHIAbwBwAHMAKAApADoAIABUACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAB0AGgAaQBzAC4AcAByAG8AcABzAAoAIAAgAH0ACgB9AAoACgBjAGwAYQBzAHMAIABNAHkAQwBsAGEAcwBzACAAZQB4AHQAZQBuAGQAcwAgAEUAbgB0AGkAdAB5ADwAewB2AGEAbAB1AGUAOgAgAHMAdAByAGkAbgBnAH0APgAgAHsACgAgACAAcwB0AGEAdABpAGMAIABjAHIAZQBhAHQAZQAoAHYAYQBsAHUAZQA6ACAAcwB0AHIAaQBuAGcAKQA6ACAATQB5AEMAbABhAHMAcwAgAHsACgAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAG4AZQB3ACAATQB5AEMAbABhAHMAcwAoAHsAIAB2AGEAbAB1AGUAIAB9ACkACgAgACAAfQAKAH0ACgAKAGMAbwBuAHMAdAAgAG0AeQBDAGwAYQBzAHMAIAA9ACAATQB5AEMAbABhAHMAcwAuAGMAcgBlAGEAdABlACgAJwBtAHkALQBjAGwAYQBzAHMAJwApAAoACgBjAG8AbgBzAG8AbABlAC4AbABvAGcAKABtAHkAQwBsAGEAcwBzAC4AcAByAG8AcABzACkA

### Expected result

I have a class with only static methods, but this class extends another class with non-static methods. Biome proposes that I replace my static method with a function. I expect no errors here as my class has non-static methods via inheritance.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
A possible fix could be to ignore classes that inherit other classes.

**Comment by Conaclos** [comment]
[kyptov](https://github.com/kyptov) just wondering: there is any practical code doing this?

> A possible fix could be to ignore classes that inherit other classes.

Yeah, this looks like a fair fix.

**Comment by kyptov** [comment]
> [kyptov](https://github.com/kyptov) just wondering: there is any practical code doing this?

- Having an empty class with different props? Yes, it helps to distinguish one type from another. But most of them are not empty. 
- A static method for creating a class? Yes, it allows you to make the constructor protected, which forces the use of the static method where some validation can be added. I like when the class and its validation reside in one place. Also, we are injecting this static method as a function, which allows us to mock it in unit tests. 
- Parent Entity class? Yes, it simplifies database interactions.
